universe/node/users.go

```diff
@@ -39,3 +40,27 @@ func (n *Node) CreateUsers(ctx context.Context, users ...*entry.User) error {
 	// with their filled in database defaults and ID.
 	return nil
 }
+
+func (n *Node) Filter(predicateFn func(userID umid.UMID, user universe.User) bool) (map[umid.UMID]universe.User, error) {
+	data := make(map[umid.UMID]universe.User)
+	userTypeID, err := common.GetNormalUserTypeID()
+	if err != nil {
+		return nil, nil
+	}
+
+	users, _ := n.db.GetUsersDB().GetAllUsers(n.ctx, userTypeID)
+
+	n.Mu.RLock()
+	defer n.Mu.RUnlock()
+
+	for _, v := range users {
+		loadedUser, _ := n.LoadUser(v.UserID)
```

Comment:
We already have all users on line 51, but here n.LoadUser will execute SQL
```getUserByIDQuery     = `SELECT * FROM "user" WHERE user_id = $1;```
for every user again